### PR TITLE
spec: Prepare openQA-devel; harmonize dependencies in cpanfile+spec

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Comment by okurz** [comment]
Testing on https://build.opensuse.org/package/show/home:okurz:branches:devel:openQA:feature:devel/openQA with the plan to use the generated container in another PR to check if it provides all dependencies.

File: openQA.spec [hunk 1]
```diff
@@ -53,6 +53,8 @@
 # all requirements needed by the tests, do not require on this in individual
 # sub-packages except for the devel package
 %define test_requires %common_requires %main_requires %python_scripts_requires %worker_requires
+%define         devel_requires %test_requires rsync ShellCheck os-autoinst-devel curl git postgresql-devel qemu qemu-kvm tar optipng postgresql-server xorg-x11-fonts rubygem(sass) sudo perl(Config) perl(Cwd) perl(Data::Dumper) perl(Digest::MD5) >= 2.55, perl(Data::OptList) perl(Devel::Cover::Report::Codecov) perl(LWP::Protocol::https) perl(SQL::SplitStatement) perl(XSLoader) >= 0.24
```
Comment (coolo) [inline openQA.spec]: there is a comma that looks suspicious - please branch

Comment (okurz) [reply openQA.spec]: branch? The comma is needed to separate every entry specifying a version comparision as already used in the macros defined above

**Comment by okurz** [comment]
https://build.opensuse.org/package/show/home:okurz:branches:devel:openQA:feature:devel/openQA looks fine meanwhile. Now I wait for all build results and see if the container is built so that we can try to use it from tests.

**Comment by okurz** [comment]
"containers" on x86_64 still gives me "unresolvable" on openQA-devel but I don't know why. Should we

a) Risk it and accept the PR?
b) Split it to have the spec and cpanfile changes in first and then in a second step change the Dockerfile?
c) Try harder to fix the container build within OBS?

**Comment by coolo** [comment]
the container so far only builds against devel:openQA:Leap and in there is no openQA-devel

**Comment by okurz** [comment]
yes, sure. This is why I added the path in the Dockerfile, e.g. see https://build.opensuse.org/package/view_file/home:okurz:branches:devel:openQA:feature:devel_container/openQA/_service:obs_scm:Dockerfile?expand=1 which then gives me "nothing provides container:opensuse:42.3", but I am not sure what that means now.

**Comment by coolo** [comment]
your repo is not of type docker - copy https://build.opensuse.org/project/prjconf/devel:openQA

**Comment by okurz** [comment]
Strange that this is not taken over correctly on branching but the linked packages including openqa_dev is

**Comment by coolo** [comment]
Feel free to file an issue on openSUSE/open-build-service - copying the project config is normally rather harmful as you link the repo you're coming from. But the repo type is special - and hiding it in the project config is a rather crude hack

**Comment by okurz** [comment]
done, thx: https://github.com/openSUSE/open-build-service/issues/8235

https://build.opensuse.org/project/show/home:okurz:branches:devel:openQA:feature:devel_container does not yet look very promising though

**Comment by coolo** [comment]
https://build.opensuse.org/package/show/home:okurz:branches:devel:openQA:feature:devel_container/openqa_dev is green. what did you change?

**Comment by okurz** [comment]
finally got it to work, mainly branching os-autoinst as well to provide os-autoinst-devel and changes in the Dockerfile:

```
+ RUN zypper ar -f -G "http://download.opensuse.org/repositories/home:/okurz:/branches:/devel:/openQA:/feature:/devel_container/openSUSE_Leap_42.3/" devel_openqa_container
+ RUN zypper ar -f -G "http://download.opensuse.org/repositories/home:/okurz:/branches:/devel:/openQA:/feature:/devel/openSUSE_Leap_42.3/" devel_openqa_test
```

so I wonder if it works if I switch to referencing the "devel:openQA" project which will provide os-autoinst-devel as well as openQA-devel.

To be able to do this change safely I have for now removed again the actual switch to "openQA-devel" and will reserve this for another PR so that we can check if all dependencies are provided within the official devel:openQA context.

https://travis-ci.org/okurz/openQA/jobs/580078577#L487 shows how the container is used within travis CI tests and as https://travis-ci.org/okurz/openQA/builds/580078576 is all green I am now confident that we can merge.

So please merge, I will have another PR doing the changes in Dockerfile.

**Pull request merged**